LSPS2/README.md

```diff
@@ -0,0 +1,736 @@
+the following:
+
+* The sum of `timestamp` and `x`/`expiry` fields is equal to
+  or less than `opening_fee_params.valid_until`.
+* `n` is the client node ID, or any public key whose private
+  key the client knows.
+* `r` contains a single routehint with a single hop:
+  * `pubkey` is the LSP public node ID.
+  * `short_channel_id` is the `jit_channel_scid`.
+  * `fee_base_msat` and `fee_proportional_millionths` are 0.
+  * `cltv_expiry_delta` is `lsp_cltv_expiry_delta`.
+* Depending on if the client specified a `payment_size` in
+  the previous `lsps2.buy` call:
+  * If `payment_size` specified ("MPP+fixed-invoice" mode),
+    set the invoice amount to `payment_size` and enable
+    the `basic_mpp` feature.
+  * If `payment_size` unspecified ("no-MPP+var-invoice" mode),
+    do not specify an invoice amount, and disable the
+    `basic_mpp` feature.
+
+All other fields are to be filled in by the client.
+
+> **Rationale** In-Lightning feerates can be folded into the
+> `opening_fee` for the initial payment.
+> The channel is not open yet, thus the client cannot know
+> what the actual feerates the LSP would impose at the time
+> this invoice is generated, so it must be fixed at this
+> point.
+> Multipart payments are specifically disallowed, as the
+> total amount of a multipart payment is encrypted in the
+> onion such that only the rceiver (the client) can decode
+> it, and the LSP cannot thus determine what the total amount
+> from the payer would be.
+> By forcing a non-multipart payment, the LSP can treat the
+> first arriving HTLC to be forwarded to the `jit_channel_scid`
+> as the total payment.
+
+#### Forward Compatibility With Post-BOLT11 Payment Schemes
+
+Ultimately, the client MUST provide the information below to
+the payer:
+
+* The client node ID (or any public key whose private key the
+  client knows).
+* The `jit_channel_scid`.
+* The LSP node ID.
+* The `lsp_cltv_expiry_delta`.
+* The expiry for the payment.
+* Whether it can use MPP or not.
+
+However, the client MAY provide this information to the payer
+by means other than BOLT11 invoice, including in a form that
+is not decodable by the payer, but which can be decoded by
+another LN participant.
+
+The only necessary requirement is for the LSP to see an incoming
+HTLC onion that indicates the `jit_channel_scid` as the next hop,
+and as long as the client is able to cause the payer to do this,
+it is would be compatible with this protocol.
+
+> **Non-normative** For example, the client node ID, the
+> `jit_channel_scid`, the LSP node ID, and the
+> `lsp_cltv_expiry_delta` can be encoded as the last hop of
+> a blinded path, for example in a BOLT12 offer or invoice.
+> The payer would not have direct access to the information,
+> but the introduction point of the blinded path would be able
+> to start the decoding of the blinded path.
+
+### 4.  Payment
+
+The client MUST issue the generated invoice to the payer of the
```

Comment:
MAY

Reply:
Right, because the client might not actually be using an invoice.

Reply:
Done.